server/data/prisonerSearchApiClient.ts

```diff
@@ -1,26 +1,31 @@
 import config, { ApiConfig } from '../config'
-import RestClient from './restClient'
 import type { Prisoner, PrisonerSearchCriteria } from '../@types/prisonerSearchApiClientTypes'
 import { PrisonerSearchByNomisIds } from '../@types/prisonerSearchApiClientTypes'
+import RestClient from './hmppsRestClient'
+import { User } from '../@types/CvlUserDetails'
 
-export default class PrisonerSearchApiClient {
-  restClient: RestClient
-
-  constructor(token: string) {
-    this.restClient = new RestClient('Prisoner search API', config.apis.prisonerSearchApi as ApiConfig, token)
+export default class PrisonerSearchApiClient extends RestClient {
+  constructor() {
+    super('Prisoner search API', config.apis.prisonerSearchApi as ApiConfig)
   }
 
-  async searchPrisoners(prisonerSearchCriteria: PrisonerSearchCriteria): Promise<Prisoner[]> {
-    return (await this.restClient.post({
-      path: '/prisoner-search/match-prisoners',
-      data: prisonerSearchCriteria,
-    })) as Promise<Prisoner[]>
+  async searchPrisoners(prisonerSearchCriteria: PrisonerSearchCriteria, user: User): Promise<Prisoner[]> {
+    return (await this.post(
+      {
+        path: '/prisoner-search/match-prisoners',
+        data: prisonerSearchCriteria,
+      },
+      { username: user.username }
+    )) as Promise<Prisoner[]>
   }
 
-  async searchPrisonersByNomsIds(nomisIdsToSearch: PrisonerSearchByNomisIds): Promise<Prisoner[]> {
-    return (await this.restClient.post({
-      path: '/prisoner-search/prisoner-numbers',
-      data: nomisIdsToSearch,
-    })) as Promise<Prisoner[]>
+  async searchPrisonersByNomsIds(nomisIdsToSearch: PrisonerSearchByNomisIds, user?: User): Promise<Prisoner[]> {
+    return (await this.post(
+      {
+        path: '/prisoner-search/prisoner-numbers',
+        data: nomisIdsToSearch,
+      },
+      { username: user?.username }
+    )) as Promise<Prisoner[]>
```

Comment:
To get here can the the `user?.username` ever be nullable?

Reply:
The Cron job uses this method, which will not pass a username